base/src/androidMain/kotlin/ApplicationHolder.kt

```diff
@@ -20,42 +20,37 @@ package com.splendo.kaluga.base
 
 import android.app.Application
 import android.content.Context
-import com.splendo.kaluga.base.ApplicationHolder.Companion.application
 
 /**
- * Class holding reference to the [Application] running Kaluga
- * set [application] to your Application so default constructors work with the proper [Context]
+ * Class holding reference to the [Application] [Context] running Kaluga
+ * set [applicationContext] to your Application so default constructors work with the proper [Context]
  */
-class ApplicationHolder {
-    companion object {
-
-        /**
-         * The [Application] running Kaluga
-         */
-        var application: Application? = null
-            set(application) {
-                check(field == null) { "Application object can only be set once." }
-                field = application
+object ApplicationHolder {
+
+    private var _applicationContext: Context? = null
```

Comment:
android system does not really use an `Application` object but relies on the `Context`

Reply:
I think it was mostly a way to use types to guarantee the context set was the actual application context and not an `Activity` context by accident